plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -382,14 +382,23 @@ export class MongoDBInstrumentation extends InstrumentationBase<
   ) {
     // add network attributes to determine the remote server
     if (topology && topology.s) {
-      span.setAttributes({
-        [SemanticAttributes.NET_HOST_NAME]: `${
-          topology.s.options?.host ?? topology.s.host
-        }`,
-        [SemanticAttributes.NET_HOST_PORT]: `${
-          topology.s.options?.port ?? topology.s.port
-        }`,
-      });
+      let host = topology.s.options?.host ?? topology.s.host;
+      let port: string | number | undefined =
+        topology.s.options?.port ?? topology.s.port;
+      if (host == null || port == null) {
+        const address = topology.description?.address;
+        if (address) {
+          const addressSegments = address.split(':');
+          host = addressSegments[0];
+          port = addressSegments[1];
+        }
+      }
+      if (host != null && port != null) {
```

Comment:
i would prefer to have stricter check there, we should explicitly check for undefined/null/empty stirng